uaclient/config.py

```diff
@@ -1024,11 +1083,55 @@ def process_config(self):
                 ).format(prop)
                 raise exceptions.UserFacingError(error_msg)
 
+        # if set on the uaclient.conf file directly
+        if getattr(self, "apt_http_proxy"):
+            if getattr(self, "global_apt_http_proxy") is None:
+                # if global_apt_* is not set yet apt_* is set then
+                # replace the former with latter
+                setattr(
+                    self,
+                    "global_apt_http_proxy",
+                    getattr(self, "apt_http_proxy"),
+                )
+            setattr(self, "apt_http_proxy", None)
+
+        if getattr(self, "apt_https_proxy"):
+            if getattr(self, "global_apt_https_proxy") is None:
+                setattr(
+                    self,
+                    "global_apt_https_proxy",
+                    getattr(self, "apt_https_proxy"),
+                )
+            setattr(self, "apt_https_proxy", None)
+
+        if getattr(self, "global_apt_http_proxy") or getattr(
+            self, "global_apt_https_proxy"
+        ):
+            if getattr(self, "ua_apt_http_proxy") or getattr(
+                self, "ua_apt_https_proxy"
+            ):
+                # Should we unset the config values?
+                raise exceptions.UserFacingError(
+                    "Warning: your configuration is not correct."
```

Comment:
Let's make this message more descriptive as well.
```suggestion
                    "Error: Setting global apt proxy and ua scoped apt proxy at the same time is unsupported. Cancelling config process operation."
```